### ISSUE TITLE
Add all required fields to the Create Dataset Form

### ISSUE THREAD

**Issue by GPortas** [IQSS/dataverse-frontend#252]
## Overview of the Feature Request

Add all required fields to the Create Dataset Form. This is only the UI, so no API connected.

![requiredfields](https://github.com/IQSS/dataverse-frontend/assets/17010447/ce59f244-17dd-40af-ac09-053125929c60)

These changes will be added to the boilerplate created on #231, so we need to wait for PR #251 to be accepted.

## What kind of user is the feature intended for?
All of them

## What inspired the request?
- https://github.com/IQSS/dataverse-frontend/issues/231

## What existing behavior do you want changed?
None

## Any brand new behavior do you want to add to Dataverse?
Include the required fields to the Create Dataset Form

## Any open or closed issues related to this feature request?
- https://github.com/IQSS/dataverse-frontend/issues/231
- https://github.com/IQSS/dataverse-frontend/issues/251

**Comment by GPortas** [comment]
Should this issue be tagged with the waiting tag until [the create dataset boilerplate PR](https://github.com/IQSS/dataverse-frontend/pull/251) is closed? I think the changes to be developed in this issue depend on the page boilerplate resulting from the PR.

@pdurbin I see you moved the issue to in progress, I'm not sure if I'm missing something.

**Comment by GPortas** [comment]
As discussed in standup.

Moved the issue back to This Sprint column and added the waiting tag due to dependency with https://github.com/IQSS/dataverse-frontend/pull/251

**Comment by pdurbin** [comment]
Weird, I don't remember moving it but a lot of issues got moved in the transition from project 2 to 34. 😅

**Comment by M27Mangan** [comment]
@GPorta, @MellyGray, @ekraffmiller  Who should I talk to regarding the dynamic creation of Form Fields, and what rules govern when certain fields are visible?

**Comment by MellyGray** [comment]
@M27Mangan, we discussed this last Monday during the post-standup meeting. I believe @GPortas had an approach for this. There are some config files that need to be read to display the form fields, and these would be obtained via API

**Comment by GPortas** [comment]
@M27Mangan 

I have moved this issue back to sprint ready, as its implementation should be wait for https://github.com/IQSS/dataverse-frontend/pull/251 to be merged.

I have created a new analysis issue, to include in this new sprint, considering your interest in yesterday's standup in analyzing the implementation of the form fields: https://github.com/IQSS/dataverse-frontend/issues/282. We don't need to wait for the analysis issue, as it does not require code changes.

I have sized the issue with a 10, but feel free to resize it as you consider.
